src/ui/public/visualize/loader/pipeline_helpers/build_pipeline.ts

```diff
@@ -41,37 +65,119 @@ const vislibCharts: string[] = [
   'line',
 ];
 
-export const getSchemas = (vis: Vis): Schemas => {
+export const getSchemas = (vis: Vis, timeRange?: any): Schemas => {
+  const createFormat = (agg: AggConfig): SchemaFormat => {
+    let format: any = agg.params.field ? agg.params.field.format.toJSON() : {};
+    if (agg.type.name === 'date_range') {
+      format = { id: 'string' };
+    } else if (agg.type.name === 'percentile_ranks') {
+      format = { id: 'percent' };
+    } else if (['count', 'cardinality'].includes(agg.type.name)) {
+      format = { id: 'number' };
+    } else if (agg.type.name === 'date_histogram') {
+      format = {
+        id: 'date',
+        params: {
+          pattern: agg.buckets.getScaledDateFormat(),
+        },
+      };
+    } else if (agg.type.name === 'terms') {
+      format.params = {
+        id: format.id,
+        otherBucketLabel: agg.params.otherBucketLabel,
+        missingBucketLabel: agg.params.missingBucketLabel,
+        ...format.params,
+      };
+      format.id = 'terms';
+    } else if (agg.type.name === 'range') {
+      format.params = { id: format.id, ...format.params };
+      format.id = 'range';
+    }
+
+    return format;
```

Comment:
The list of conditionals could potentially be simplified here by creating a map of `agg.type.name`s with corresponding formats, similar to `buildPipelineVisFunction` below.

Something along the lines of:
```typescript
const format: any = agg.params.field ? agg.params.field.format.toJSON() : {};
const formats = {
  date_range: { id: 'string' },
  percentile_ranks: { id: 'percent' },
  count: { id: 'number' },
  cardinality: { id: 'number' },
  date_histogram: {
    id: 'date',
    params: {
      pattern: agg.buckets.getScaledDateFormat(),
    },
  },
  terms: {
    id: 'terms',
    params: {
      id: format.id,
      otherBucketLabel: agg.params.otherBucketLabel,
      missingBucketLabel: agg.params.missingBucketLabel,
      ...format.params,
    }
  },
  range: {
    id: 'range',
    params: { id: format.id, ...format.params },
  }
};

return formats[agg.type.name] || format;
```